picketlink/src/main/java/org/wildfly/extension/picketlink/idm/model/PartitionManagerAddHandler.java

```diff
@@ -92,398 +41,4 @@ protected void populateModel(ModelNode operation, ModelNode model) throws Operat
         }
     }
 
-    @Override
-    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
-        PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
-        final String federationName = address.getLastElement().getValue();
-        ModelNode partitionManager = Resource.Tools.readModel(context.readResource(EMPTY_ADDRESS));
-        createPartitionManagerService(context, federationName, partitionManager, false);
-    }
-
-    public void validateModel(final OperationContext context, String partitionManagerName, final ModelNode partitionManager) throws OperationFailedException {
-        createPartitionManagerService(context, partitionManagerName, partitionManager, true);
-    }
-
-    public void createPartitionManagerService(final OperationContext context, String partitionManagerName, final ModelNode partitionManager, boolean onlyValidate) throws OperationFailedException {
-        String jndiName = PartitionManagerResourceDefinition.IDENTITY_MANAGEMENT_JNDI_URL
-            .resolveModelAttribute(context, partitionManager).asString();
-        IdentityConfigurationBuilder builder = new IdentityConfigurationBuilder();
-        PartitionManagerService partitionManagerService = new PartitionManagerService(partitionManagerName, jndiName, builder);
-        ServiceBuilder<PartitionManager> serviceBuilder = null;
-
-        if (!onlyValidate) {
-            serviceBuilder = context.getServiceTarget()
-                .addService(PartitionManagerService.createServiceName(partitionManagerName), partitionManagerService);
-        }
-
-        ModelNode identityConfigurationNode = partitionManager.get(IDENTITY_CONFIGURATION.getName());
-
-        if (!identityConfigurationNode.isDefined()) {
-            throw ROOT_LOGGER.idmNoIdentityConfigurationProvided();
-        }
-
-        for (Property identityConfiguration : identityConfigurationNode.asPropertyList()) {
-            String configurationName = identityConfiguration.getName();
-            NamedIdentityConfigurationBuilder namedIdentityConfigurationBuilder = builder.named(configurationName);
-
-            if (!identityConfiguration.getValue().isDefined()) {
-                throw ROOT_LOGGER.idmNoIdentityStoreProvided(configurationName);
-            }
-
-            List<ModelNode> identityStores = identityConfiguration.getValue().asList();
-
-            for (ModelNode store : identityStores) {
-                configureIdentityStore(context, serviceBuilder, partitionManagerService, configurationName, namedIdentityConfigurationBuilder, store);
-            }
-        }
```

Comment:
No, this is part of createPartitionManagerService() method, which does also some minor validation but 99% of time is used in RUNTIME stage.
I've extracted the model validation part in method named validatePartitionManager() that is now used for model validation when needed.